pyzebra/fit2.py

```diff
@@ -0,0 +1,225 @@
+            x, result.params["g_cen"].value + 3 * result.params["g_width"].value
+        )
+        numfit_min = gauss_3sigmamin if numfit_min is None else find_nearest(x, numfit_min)
+        numfit_max = gauss_3sigmamax if numfit_max is None else find_nearest(x, numfit_max)
+
+        it = -1
+        while numfit_max == numfit_min:
+            # in the case the peak is very thin and numerical integration would be on zero omega difference, finds closes values
+            it = it + 1
+            numfit_min = find_nearest(
+                x,
+                result.params["g_cen"].value - 3 * (1 + it / 10) * result.params["g_width"].value,
+            )
+            numfit_max = find_nearest(
+                x,
+                result.params["g_cen"].value + 3 * (1 + it / 10) * result.params["g_width"].value,
+            )
+
+        if x[numfit_min] < np.min(x):
+            # makes sure that the values supplied by user lay in the omega range
+            # can be ommited for users who know what they're doing
+            numfit_min = gauss_3sigmamin
+            print("Minimal integration value outside of x range")
+        elif x[numfit_min] >= x[numfit_max]:
+            numfit_min = gauss_3sigmamin
+            print("Minimal integration value higher than maximal")
+        else:
+            pass
+        if x[numfit_max] > np.max(x):
+            numfit_max = gauss_3sigmamax
+            print("Maximal integration value outside of x range")
+        elif x[numfit_max] <= x[numfit_min]:
+            numfit_max = gauss_3sigmamax
+            print("Maximal integration value lower than minimal")
+        else:
+            pass
+
+        count_errors = create_uncertanities(y, y_err)
+        # create error vector for numerical integration propagation
+        num_int_area = simps(count_errors[numfit_min:numfit_max], x[numfit_min:numfit_max])
+        slope_err = u.ufloat(result.params["slope"].value, result.params["slope"].stderr)
+        # pulls the nominal and error values from fit (slope)
+        intercept_err = u.ufloat(
+            result.params["intercept"].value, result.params["intercept"].stderr
+        )
+        # pulls the nominal and error values from fit (intercept)
+
+        background_errors = np.array([])
+        for j in range(len(x[numfit_min:numfit_max])):
+            # creates nominal and error vector for numerical integration of background
+            bg = slope_err * (x[j] - centre) + intercept_err
+            background_errors = np.append(background_errors, bg)
+
+        num_int_background = simps(background_errors, x[numfit_min:numfit_max])
+        int_area = num_int_area - num_int_background
+
+    d = {}
+    for pars in result.params:
+        d[str(pars)] = (result.params[str(pars)].value, result.params[str(pars)].vary)
+    print(result.fit_report())
+
+    print((result.params["g_amp"].value - int_area.n) / result.params["g_amp"].value)
+    d["export_fit"] = False

```

Comment:
This parameter is (will be) used by export comm to decide if the fitted area or numerically integrated area is to be exported. User has to be able to choose, non the less numerically integrated area is default.

Reply:
Does it need to be individual for each measurement? I would expect a user wanting to export all measurements in a similar way, thus having this parameter global.

Reply:
Generally, user wants to export all fitted or integrated, but option to select individually in case of bad fit should be kept. However, I dont expect it to be used very often.